### PR TITLE
Migrate to Java HTTP Client and optimize dependencies

### PR DESCRIPTION
**Pull request by novotnyr**
Dependencies consolidation:
- replace GSON with Jackson
- remove Kotson
- remove commons-lang3
- remove OkHttp explicit dependency
- replace Retrofit with Java 11+ HTTP client

Migrate Retrofit usages to Java HTTP client.
Add unit tests against Mock Server.

### CODE REVIEW

File: intellij-plugin-verifier/verifier-repository/src/main/java/com/jetbrains/pluginverifier/repository/downloader/UrlDownloader.kt [hunk 1]
```diff
@@ -143,18 +93,88 @@ class UrlDownloader<in K>(private val urlProvider: (K) -> URL?) : Downloader<K>
...
+    }
+
+    private fun assertHttpOk(url: String, response: HttpResponse<InputStream>) {
+      when (val code = response.statusCode()) {
+        200 -> return
+        404 -> throw NotFound404ResponseException(url)
+        500 -> throw ServerInternalError500Exception(url)
+        503 -> throw ServerUnavailable503Exception(url)
+        else -> {
+          val message = response.body().bufferedReader().readLine().take(100)
```
Comment (YannCebron) [inline intellij-plugin-verifier/verifier-repository/src/main/java/com/jetbrains/pluginverifier/repository/downloader/UrlDownloader.kt:127]: why '100'? might be too little?

File: intellij-plugin-verifier/verifier-repository/src/main/java/com/jetbrains/pluginverifier/repository/downloader/UrlDownloader.kt [hunk 2]
```diff
@@ -143,18 +93,88 @@ class UrlDownloader<in K>(private val urlProvider: (K) -> URL?) : Downloader<K>
...
+          throw NonSuccessfulResponseException(url, code, message)
+        }
+      }
+    }
   }
+}
+
+private fun HttpResponse<*>.contentLength(): Long {
+  return this.headers()
+          .firstValueAsLong("Content-Length")
```
Comment (YannCebron) [inline intellij-plugin-verifier/verifier-repository/src/main/java/com/jetbrains/pluginverifier/repository/downloader/UrlDownloader.kt:137]: -> HttpHeaders?

File: intellij-plugin-verifier/verifier-repository/src/main/java/com/jetbrains/pluginverifier/misc/RestApis.kt [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+import org.apache.commons.compress.compressors.xz.XZCompressorInputStream
+import org.bouncycastle.cms.CMSSignedData
+import java.net.URI
+import java.net.http.HttpClient
+import java.net.http.HttpRequest
+import java.net.http.HttpResponse
+import java.time.Duration
+import java.util.concurrent.Executors
+
+fun createHttpClient(timeout: Duration = Duration.ofMinutes(5)): HttpClient {
```
[the remark this answers is not in the document]
Comment (novotnyr) [reply intellij-plugin-verifier/verifier-repository/src/main/java/com/jetbrains/pluginverifier/misc/RestApis.kt:17]: 5 minutes was the previous default. I picked that from the previous implementation. There was neither any configuration nor overriding of this default in the code.

File: intellij-plugin-verifier/verifier-repository/src/main/java/com/jetbrains/pluginverifier/misc/RestApis.kt [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+  return HttpClient.newBuilder().connectTimeout(timeout)
+          .executor(Executors.newCachedThreadPool(
+                  ThreadFactoryBuilder()
+                          .setNameFormat("ok-http-thread-%d")
```
[the remark this answers is not in the document]
Comment (novotnyr) [reply intellij-plugin-verifier/verifier-repository/src/main/java/com/jetbrains/pluginverifier/misc/RestApis.kt:21]: I don't get this hint - can you elaborate?

Picked from the previous implementation.

File: intellij-plugin-verifier/verifier-repository/src/main/java/com/jetbrains/pluginverifier/misc/RestApis.kt [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+                          .setDaemon(true)
+                          .build()
+          ))
+          .followRedirects(HttpClient.Redirect.NORMAL)
+          .build()
+}
+
+class RestApis {
```
[the remark this answers is not in the document]
Comment (novotnyr) [reply intellij-plugin-verifier/verifier-repository/src/main/java/com/jetbrains/pluginverifier/misc/RestApis.kt:29]: 5 minutes was the previous default. I picked that from the previous implementation. There was neither any configuration nor overriding of this default in the code.

File: intellij-plugin-verifier/verifier-repository/src/main/java/com/jetbrains/pluginverifier/misc/RestApis.kt [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+  private val httpClient = createHttpClient()
+  private val jackson = jacksonObjectMapper()
+          .configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)
+
+  fun getRawString(url: String, timeout: Duration = Duration.ofMinutes(5)): RestApiResult<String> {
+    val request = get(url, timeout)
+
+    val response = httpClient.send(request, HttpResponse.BodyHandlers.ofString())
+    if (response.statusCode() != 200) {
```
[the remark this answers is not in the document]
Comment (novotnyr) [reply intellij-plugin-verifier/verifier-repository/src/main/java/com/jetbrains/pluginverifier/misc/RestApis.kt:38]: Redirects are handled automatically (`.followRedirects(HttpClient.Redirect.NORMAL)`). 

Added unit test to verify that.

File: intellij-plugin-verifier/verifier-repository/src/main/java/com/jetbrains/pluginverifier/repository/downloader/UrlDownloader.kt (hunk 1 above)
Comment (novotnyr) [reply intellij-plugin-verifier/verifier-repository/src/main/java/com/jetbrains/pluginverifier/repository/downloader/UrlDownloader.kt]: Increased to good ole' 255.

File: intellij-plugin-verifier/verifier-repository/src/main/java/com/jetbrains/pluginverifier/misc/RestApis.kt (hunk 4 above)
[the remark this answers is not in the document]
Comment (YannCebron) [reply intellij-plugin-verifier/verifier-repository/src/main/java/com/jetbrains/pluginverifier/misc/RestApis.kt:21]: why is it called "ok-http-thread" and not  "plugin-verifier-thread" or similar?

**Review by YannCebron — APPROVED** [review]

**Pull request merged**